### ISSUE TITLE
PG3 installs to a "hidden" directory.

### ISSUE THREAD

**Issue by jimboca** [UniversalDevicesInc/pg3#55]
Was discussed on Slack but moving here for a final answer.

pg3 is installed to /var/polyglot/.pg3 where /var/polyglot already contains pg2 stuff.  Should it be changed to /var/pg3 or /var/polyglot/pg3 

@mkohanim agreed it should be hidden and was going to talk to @exking about changing it.

Would like this resolved along with #28 before I start running pg3 nodeservers on my production system.

**Comment by bpaauwe** [comment]
The PG3 default working directory is ~/.pg3
The PG2 default working directory is ~/.polyglot

For both, it is possible to override the default.  PG2 on Polisy overrides the default to place the files in the /var/polyglot where /var/polyglot is the home directory for the polyglot user account.
Currently, the PG3 install on Polisy is not overriding the default, thus using the default of /var/polyglot/.pg3 (~/.pg3)

Thus there are two different ways the install directory can be changed.

1. Continue to keep the same design as PG2 and override the default install location.  This would be Polisy packaging change, not a PG3 change.
2. Change the PG3 default install location to a different value in the PG3 code. No packaging change would be needed.

I don't see any reason not to change the default, but since I wasn't involved in the original design decision to use ~/.polyglot (and ~/.pg3), I don't feel comfortable changing it without some input from those that made the original decision.

**Comment by jimboca** [comment]
In my opinion the original . polyglot was because it was installed as current user running on RPi or whatever they were using. With Polisy this is not the case, right?

**Comment by mkohanim** [comment]
Currently,/var/polyglot has its own filesystem in zfs for 10G. In order not to have to repartition, or put in /var I think it should stay in the /var/polyglot directory. I don't really know what are the advantages of it being hidden.

With kind regards,

******************************
Michel Kohanim

(p) 818.631.0333
http://www.universal-devices.com
******************************

________________________________

**Comment by bpaauwe** [comment]
On Polisy, the user is 'polyglot' and by default PG2 would install in /var/polyglot/.polyglot but it is overriden to install in /var/polyglot

Since PG3 is supposed to install on Polisy only, I don't see any reason why it we shouldn't change the default to something like 'pg3' so it installs in /var/polyglot/pg3 by default.

The only implication is that it won't carry over any existing configuration.  It will be a new install in a new location when upgrading  with a new default directory set.  I'll go ahead and change the default to 'pg3' for the next release.

**Comment by jimboca** [comment]
Thanks @bpaauwe 
Seems good to make the change now before it's too late :)

**Comment by bpaauwe** [comment]
I just realized as I made the change, that it's going to mess up my PG3 development because I have my clone of the pg3 git repository in ~/pg3 and it will now put all of the configuration/runtime files in there as well.  I'm going to have to move my clone of the repository to a different location.  So I guess this is one down side of using 'pg3' as the default but currently, that only effects me.

**Comment by jimboca** [comment]
Yes, and only if you do development as the polyglot user, right?
